File: LeetCode/Array/RemoveElement.swift
//
//  RemoveElement.swift
//  LeetCode
//
//  Created by chaox on 2022/6/15.
//  Copyright © 2022 ios. All rights reserved.
//

import UIKit

//27移除元素
func removeElement(_ nums: inout [Int], _ val: Int) -> Int {

    //快慢指针
    var slowIndex = 0
    for (fastIndex,value) in nums.enumerated() {
        if value != val {
            nums[slowIndex] = nums[fastIndex]
            slowIndex += 1
        }
    }
    
    return slowIndex
}

File: LeetCode/Array/RemoveElement.cpp
//
//  RemoveElement.cpp
//  LeetCode
//
//  Created by 潮汐 on 2022/8/13.
//  Copyright © 2022 ios. All rights reserved.
//

#include "Header.h"

//27. 移除元素
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        
        
        int fast = 0;
        int slow = 0;
        
        while (fast < nums.size()) {
            if (nums[fast] != val) {
                nums[slow] = nums[fast];
                slow++;
            }
            fast++;
        }
        
        return slow;
    }
};
